source/OrderedTypes/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,2209 @@
+
+ retraction-T'-to-T : T-to-T' ∘ T'-to-T ∼ id
+ retraction-T'-to-T = inverses-are-retractions' T'-≃-T
+
+ T'-inclusion : T' → ⟨ L ⟩
+ T'-inclusion = m ∘ T'-to-T
+
+ s : ⟨ L ⟩
+ s = ⋁ (T' , T'-inclusion)
+
+ open Joins _≤_
+
+ sup-of-small-fam-is-lub : (s is-lub-of (T , m)) holds
+ sup-of-small-fam-is-lub = (s-upper-bound , s-least-upper-bound)
+  where
+   s-upper-bound : (s is-an-upper-bound-of (T , m)) holds
+   s-upper-bound t = t-≤-s
+    where
+     t-≤-s : (m t ≤ s) holds 
+     t-≤-s = transport (λ z → (m z ≤ s) holds) (section-T'-to-T t)
+                       ((is-an-upper-bound-for L of
+                        (T' , T'-inclusion)) (T-to-T' t))
+   s-least-upper-bound : ((u , _) : upper-bound (T , m)) → (s ≤ u) holds
+   s-least-upper-bound (u , is-upbnd-T) = s-≤-u
+    where
+     s-≤-u : (s ≤ u) holds
+     s-≤-u = pr₂ (is-lub-for L (T' , T'-inclusion))
+                 ((u , λ i → is-upbnd-T (T'-to-T i)))
+
+\end{code}
+
+We now show that reindexing across an equivalence does not change the join.
+
+\begin{code}
+
+module equivalent-families-have-same-join {𝓤 𝓦 𝓥 𝓣 𝓣' : Universe}
+                                          (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                                          (T : 𝓣  ̇)
+                                          (T' : 𝓣'  ̇)
+                                          (e : T' ≃ T)
+                                          (m : T → ⟨ L ⟩)
+                                          where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+
+ reindexing-along-equiv-＝-sup : (s s' : ⟨ L ⟩)
+                               → (s is-lub-of (T , m)) holds
+                               → (s' is-lub-of (T' , m ∘ ⌜ e ⌝ )) holds
+                               → s ＝ s'
+ reindexing-along-equiv-＝-sup s s' (is-upbnd , is-least-upbnd)
+                        (is-upbnd' , is-least-upbnd') =
+   is-antisymmetric-for L s-≤-s' s'-≤-s
+  where
+   s-≤-s' : (s ≤ s') holds
+   s-≤-s' = is-least-upbnd (s' , λ t → transport (λ z → (z ≤ s') holds)
+                           (＝₁ t) (is-upbnd' (⌜ e ⌝⁻¹ t)))
+    where
+     ＝₁ : (t : T) → m (⌜ e ⌝ (⌜ e ⌝⁻¹ t)) ＝ m t
+     ＝₁ t = ap m (naive-inverses-are-sections ⌜ e ⌝ (⌜ e ⌝-is-equiv) t)
+   s'-≤-s : (s' ≤ s) holds
+   s'-≤-s = is-least-upbnd' (s , λ t' → is-upbnd (⌜ e ⌝ t'))
+
+\end{code}
+
+We can weaken the above result to a surjection.
+
+\begin{code}
+
+module surjection-implies-equal-joins {𝓤 𝓦 𝓥 𝓣 𝓣' : Universe}
+                                      (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                                      (T : 𝓣  ̇)
+                                      (T' : 𝓣'  ̇)
+                                      (e : T' ↠ T)
+                                      (m : T → ⟨ L ⟩)
+                                      where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+ open PropositionalTruncation pt
```

Comment:
You should only open this **once** at the top of the file, just after the top-level module declaration.